grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -673,53 +674,40 @@ class ProductAvailabilityService {
             }
         }
 
-        def products = Product.createCriteria().list {
-            eq("active", true)
-            and {
-                if (categories) {
-                    'in'("category", categories)
-                }
-                if (command.tags) {
-                    tags {
-                        'in'("id", command.tags*.id)
+        return ProductSearch.createCriteria().list(max: command.maxResults, offset: command.offset) {
+            product {
+                eq("active", true)
+                and {
+                    if (categories) {
+                        'in'("category", categories)
                     }
-                }
-                if (command.catalogs) {
-                    productCatalogItems {
-                        productCatalog {
-                            'in'("id", command.catalogs*.id)
+                    if (command.tags) {
+                        tags {
+                            'in'("id", command.tags*.id)
                         }
                     }
+                    if (command.catalogs) {
+                        productCatalogItems {
+                            productCatalog {
+                                'in'("id", command.catalogs*.id)
+                            }
+                        }
+                    }
+                    // This is pretty inefficient if the previous query does not narrow the results
+                    // if the inner products list is empty, but there are search terms then return empty results
+                    if (innerProductIds || searchTerms) {
+                        'in'("id", innerProductIds ?: [null])
+                    }
                 }
-                // This is pretty inefficient if the previous query does not narrow the results
-                if (innerProductIds) {
-                    'in'("id", innerProductIds)
+            }
+            eq("location", command.location)
+            or {
+                isNull("type")
+                and {
+                    eq("isSearchableType", Boolean.TRUE)
+                    gt("quantityOnHand", 0)
```

Comment:
I think we should leave this up to the caller. There are negative quantities that we probably want to return in some cases (i.e. data error detection)

Reply:
This is the initial request from the ticket related to the searchable product types (if the type is searchable, show only products that have QoH)

Reply:
Ok.